x-pack/plugins/enterprise_search/server/lib/fetch_indices.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IScopedClusterClient } from 'kibana/server';
+
+import { ByteSizeValue } from '@kbn/config-schema';
+
+async function fetchIndicesCall(client: IScopedClusterClient): Promise<unknown> {
+  const indexNamesString = 'enterprise-search-index-*';
+
+  // This call retrieves alias and settings (incl. hidden status) information about indices
+  const indices = await client.asCurrentUser.indices.get({
+    index: indexNamesString,
+    expand_wildcards: ['open'],
+    // only get specified index properties from ES to keep the response under 536MB
+    // node.js string length limit: https://github.com/nodejs/node/issues/33960
+    filter_path: [
+      '*.aliases',
+      '*.settings.index.number_of_shards',
+      '*.settings.index.number_of_replicas',
+      '*.settings.index.frozen',
+      '*.settings.index.hidden',
```

Comment:
I guess we don't have to return the `hidden` setting because the `expand_wildcards: ['open']` will restrict the result set to only open, non-hidden indices (based on my reading of https://www.elastic.co/guide/en/elasticsearch/reference/8.0/indices-get-settings.html#get-index-settings-api-query-params ).